src/drive/web/modules/drive/Toolbar/components/Scanner.jsx

```diff
@@ -0,0 +1,210 @@
+                )
+                //It's possible that the filename as changed, so let's update it
+                if (onConflict === 'rename') {
+                  this.setState({ filename: newFile.data.name })
+                }
+
+                if (onFinish) onFinish()
+              } catch (error) {
+                this.setState({ error })
+              } finally {
+                this.setState({ status: 'done' })
+              }
+            }
+            // Read the file as an ArrayBuffer
+            reader.readAsArrayBuffer(file)
+          },
+          err => {
+            this.setState({ error: err })
+            this.setState({ status: 'done' })
+            console.error('error getting fileentry file!' + err)
+          }
+        )
+      },
+      error => {
+        this.setState({ error })
+        this.setState({ status: 'done' })
+      }
+    )
+  }
+
+  //TODO Put this in Files Doctypes
+  /**
+   *
+   * @param {String} name File Name
+   * @param {ArrayBuffer} file data
+   * @param {String} dirId dir id when to upload
+   * @param {String} onConflict Actually only 2 hardcoded strategies 'erase' or 'rename'
+   */
+  async uploadFileWithConflictStrategy(name, file, dirId, onConflict) {
+    const { client } = this.props
+    const filesCollection = client.collection('io.cozy.files')
+
+    const path = await CozyFile.getFullpath(dirId, name)
+    try {
+      const existingFile = await filesCollection.statByPath(path)
+      const { id: fileId, dir_id: dirId } = existingFile.data
+      if (onConflict === 'erase') {
+        //!TODO Bug Fix. Seems we have to pass a name attribute ?!
+        const resp = await filesCollection.updateFile(file, {
+          dirId,
+          fileId,
+          name
+        })
+        return resp
+      } else {
+        const { filename, extension } = CozyFile.splitFilename({
+          name,
+          type: 'file'
+        })
+        const newFileName =
+          CozyFile.generateNewFileNameOnConflict(filename) + extension
+        //recall itself with the newFilename.
+        return this.uploadFileWithConflictStrategy(
+          newFileName,
+          file,
+          dirId,
+          onConflict
+        )
+      }
+    } catch (error) {
+      if (/Not Found/.test(error)) {
+        return await this.upload(name, file, dirId)
+      }
+      throw error
+    }
+  }
+  async upload(name, file, dirId) {
+    const { client } = this.props
+
+    return client
+      .collection('io.cozy.files')
+      .createFile(file, { name, dirId, contentType: 'image/jpeg' })
+  }
+  onFail(message) {
+    console.log('failed', message)
+  }
+  onClick = () => {
+    this.defaultPluginConfig = {
+      quality: 70,
+      destinationType: window.navigator.camera.DestinationType.FILE_URI,
+      sourceTypes: window.navigator.camera.PictureSourceType.CAMERA,
+      correctOrientation: true
+    }
+
+    window.navigator.camera.getPicture(this.onSuccess, this.onFail, {
```

Comment:
We should have a try/catch here or something, in case the plugin is not available.